TeamProductionSystem/Team_Production_System_BE
# Feat/issue 154/configuring linting

## **Pull Request Template**

### **1. Targeted Issue** ###

*(The first step to a good solution is knowing the problem. Please link the issue here.)*

Issue #... or
Ticket #...

https://trello.com/c/UIf8r7Yr/154-add-additional-linting-plugins-to-improve-overall-code-consistency

### **2. Overview of Solution** ###

*(You've put in the hard work, now's your chance to tell us about it. What's your solution in a nutshell?)*

I've configured linting and auto-formatting to be performed whenever a contributor tries to make a commit. I've also added documentation detailing the linting plugins and how to setup pre-commit.


### **3. Tools Used** ###

*(Share the tools, libraries, or techniques you've used in your solution.)*

pre-commit, flake8, isort, black


### **4. Testing Strategy** ###

*(No code is complete without tests. Outline what you've tested and how.)*

1) Make a change to the codebase that is technically correct, but difficult to read.
2) Try to make a commit with a message that doesn't follow the schema in the docs.

You should see that the commit fails the isort, black, or flake8 steps, and fails the commit message step.
If isort or black fail, there will be new changes to commit.
In the end, the commit is cancelled.


### **5. Future Implications** ###

*(Let's anticipate potential effects of your changes. Are there any potential issues down the line?)*

Contributors will now have to follow the code style enforced by the linter and auto-formatters, leading to a more consistent codebase. If we want to require type hints in the future, we can use a flake8 plugin to include rules in the linter.


### **6. Screenshots** ###
<img width="711" alt="Screenshot 2023-10-25 at 1 48 06 PM" src="https://github.com/TeamProductionSystem/Momentors_Backend/assets/8958872/6d4c3cb6-54e0-49b2-aa3d-4238d6cad2cb">
<img width="711" alt="Screenshot 2023-10-25 at 1 48 42 PM" src="https://github.com/TeamProductionSystem/Momentors_Backend/assets/8958872/c300f839-e92b-4314-bdeb-90f8cc1baf46">

*(A picture is worth a thousand words. Include screenshots of your changes if relevant.)*


### **7. Code Reviewers** ###

*(Who else is in the loop? Mention your team or other contributors here.)*

@esparr @GitLukeW 


---

Before you go...

- [x ]  I have linked the relevant issue
- [x ]  I have provided an overview of my solution
- [x ]  I have shared the tools I've used
- [x ]  I have outlined my testing strategy
- [x ]  I have considered future implications of my changes
- [x ]  I have included relevant screenshots
- [x ]  I have kept my team in the loop

---

### Big high-fives for your incredible pull request! We're thrilled to have you as a contributor! Our team will dig into your proposed changes and give you some feedback or merge them if they're good to go. Your contributions mean the world to us!
